### PR TITLE
Add deriveCredential operation

### PR DESCRIPTION
**Pull request by OR13**
This operation is used to derive a credential for selective disclosure from another verifiable credential.

### CODE REVIEW

**Comment by OR13** [comment]
ping @tplooker

**Review by mprorock — APPROVED** [review]
This is awesome - we have a definite case for derived versions of ag inspection credentials where certain items are witheld from certain audiences/receivers

File: docs/vc-http-api.yml [hunk 1]
```diff
@@ -41,6 +41,32 @@ paths:
           description: invalid input!
         "500":
           description: error!
+  /credentials/derive:
+    post:
+      tags:
+        - Holder
+      summary: Derive a credential and returns it in the response body.
```
Comment (TallTed) [inline docs/vc-http-api.yml:48]:
```suggestion
      summary: Derives a credential and returns it in the response body.
```

File: docs/vc-http-api.yml [hunk 2]
```diff
@@ -134,11 +160,11 @@ paths:
               schema:
                 $ref: "#/components/schemas/VerifyPresentationResponse"
         "400":
-          description: Invalid or malformed input 
+          description: Invalid or malformed input
         "413":
           description: Payload too large
         "429":
-          description: Rate limit exceeded by too many requests. 
+          description: Rate limit exceeded by too many requests.
```
Comment (TallTed) [inline docs/vc-http-api.yml:167]:
```suggestion
          description: Request rate limit exceeded.
```

**Review by tplooker — APPROVED** [review]

**Pull request merged**